src/migrateTable.ts

```diff
@@ -65,11 +67,17 @@ export function migrateTable(
 
   const legacyColumns =
     legacyTableState.value?.body?.configuration?.tabular?.columns;
+
+  const columnsAxis = query.mdx?.axes[0];
+  const columnLevels = columnsAxis ? getLevels(columnsAxis, { cube }) : [];
+
   const columnWidths = legacyColumns
     ? _migrateTableColumnWidths({
         legacyColumns,
         mapping,
         cube,
+        maxLevelDepth: columnLevels.length,
```

Comment:
I believe that the computation of the max level depth is a bit more complex as it should include the index of each level within its parent hierarchy. For instance in the sandbox, if `columnLevels` contains a single element which is `BookId`, then the max level depth is 4 and not 1. Indeed, the column contains legal entities, business units, desks and finally book ids.

Reply:
My latest commits address this comment 👍